erpnext/stock/doctype/batch/batch.py

```diff
@@ -393,6 +393,62 @@ def get_readings_for_qi(qi_name):
 	readings_info = {reading.specification: reading.reading_1 for reading in readings}
 	return readings_info
 
+def get_batch_reserved_qty(item_code, warehouse, batch_no):
+	"""
+	Returns the reservec quantity for a particular batched item.
+
+	Args:
+		item_code (string): the item code of the item we are seeking the reserved quantity for 
+		warehouse (string): the warehouse in which the item is stored in
+		batch_no (string): the batch number of the item
+
+	Returns:
+		float: the reserved quantity of the batch
+	"""	
+	reserved_qty = frappe.db.sql("""
+		select
+			sum(dnpi_qty * ((so_item_qty - so_item_delivered_qty) / so_item_qty))
+		from
+			(
+				(select
+					qty as dnpi_qty,
+					(
+						select qty from `tabSales Order Item`
+						where name = dnpi.parent_detail_docname
+						and (delivered_by_supplier is null or delivered_by_supplier = 0)
+					) as so_item_qty,
+					(
+						select delivered_qty from `tabSales Order Item`
+						where name = dnpi.parent_detail_docname
+						and delivered_by_supplier = 0
+					) as so_item_delivered_qty,
+					parent, name
+				from
+				(
+					select qty, parent_detail_docname, parent, name
+					from `tabPacked Item` dnpi_in
+					where item_code = %s and warehouse = %s
+					and parenttype="Sales Order"
+					and item_code != parent_item
```

Comment:
- Rename to `dnpi` to something better
- Prepend table name to fields, there are chances that mysql might throw ambiguous error .

Reply:
I've just copied this from the reserve quantity function for stock without batches.

Reply:
Then you'll need to fix it over there as well. Just add what is `dnpi_qty` to the docstring above, its okay to use acronyms in large statements but it should be defined somewhere.